EMQ X Broker v4.2.x failed to build

#### Environment

- OS: macos 10.15.7
- Erlang/OTP: 21.3.8.18
- EMQ: 4.2.0/1/2

#### Description

Install from source:

`
$ git clone -b v4.2.2 https://github.com/emqx/emqx-rel.git emqx-4.2.2
$ cd emqx-4.2.2
$ export EMQX_DEPS_DEFAULT_VSN=v4.2.2
$ make
`

I failed with the following output:
`
export EMQX_DESC="EMQ X Broker"
/Users/eamon/dev/emqx-4.2.2/rebar3 as emqx release
===> Error evaluating configuration script at "rebar.config.script":
{error,{217,file,
        {error,badarg,
               [{re,run,
                    [[40,38750,20998,25903,41],
                     "master|^dev/|^hotfix/",
                     [{capture,none}]],
                    []},
                {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},
                {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,273}]},
                {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,449}]},
                {erl_eval,exprs,5,[{file,"erl_eval.erl"},{line,126}]},
                {file,eval_stream2,6,[{file,"file.erl"},{line,1393}]},
                {file,script,2,[{file,"file.erl"},{line,1090}]},
                {rebar_config,consult_and_eval,2,
                              [{file,"/Users/emqer/code/rebar3/src/rebar_config.erl"},
                               {line,287}]}]}}}

===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace or consult rebar3.crashdump
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
make: *** [emqx] Error 1
`

(base) ➜  emqx-4.2.2 git:(4b85617) ✗ make
export EMQX_DESC="EMQ X Broker"
/Users/eamon/dev/emqx-4.2.2/rebar3 as emqx release
===> Error evaluating configuration script at "rebar.config.script":
{error,{217,file,
        {error,badarg,
               [{re,run,
                    [[40,38750,20998,25903,41],
                     "master|^dev/|^hotfix/",
                     [{capture,none}]],
                    []},
                {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},
                {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,273}]},
                {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,449}]},
                {erl_eval,exprs,5,[{file,"erl_eval.erl"},{line,126}]},
                {file,eval_stream2,6,[{file,"file.erl"},{line,1393}]},
                {file,script,2,[{file,"file.erl"},{line,1090}]},
                {rebar_config,consult_and_eval,2,
                              [{file,"/Users/emqer/code/rebar3/src/rebar_config.erl"},
                               {line,287}]}]}}}

===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace or consult rebar3.crashdump
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
make: *** [emqx] Error 1

--- a/.ci/build_packages/Dockerfile
+++ b/.ci/build_packages/Dockerfile
@@ -7,6 +7,6 @@ COPY . /emqx-rel
 
 WORKDIR /emqx-rel
 
-RUN make ${EMQX_NAME}-pkg
+RUN sed -i "/^export LC_ALL=.*$/d" Makefile && make ${EMQX_NAME}-pkg
 
 RUN /emqx-rel/.ci/build_packages/tests.sh

--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,7 @@
 
 REBAR_GIT_CLONE_OPTIONS += --depth 1
 export REBAR_GIT_CLONE_OPTIONS
+export LC_ALL=en_US.UTF-8
 
 REBAR = $(CURDIR)/rebar3
 

